### DISCUSSION TITLE
Poor video performance

### DISCUSSION THREAD

**Discussion by shadow82** [photoprism/photoprism#3345] [Q&A] [answered]
Hi!
I started to imprt and test my photo albums, they look really great!
Still I did hit an issue with video performance of video when watching it via Internet.The Video is: H264 - MPEG-4 AVC FullHD 30fps video
When the Client starts to watch a video via Internet, the video chunks
I don't think my homeserver nor smartphone may be a bottleneck (Phone: S23 Ultra, Server: 2x Xeon CPU with 12  2.6GHz cores and 144 GB RAM + HDD that has HDTune results of: 22/70/155 min/avg/max MB/s read test
Server speedtest results: Down: 579Mb/s, Up: 204Mb/sPhone speedtest results:  Down 150Mb/s, Up: 55Mb/s
The thing that bothers me is netwrk performance.When server sends the video, the performance gets around 40Mb/s which for 30 fps FullHD video is enormously high.
When I was testing IPTV I barely - really barely saw usage ~20Mb/s when there was music video from some discoteque (much is happening on the screen, so the codec has to send more data at time).
And here - 30fps FullHD really static video and the usage is around 40Mb/s

I have installed Photoprism preview version and I have installed ffmpeg 4.2.7 (if it's needed for this kind of video). Everything is new and fresh - 2 days only
What can I do to make PhotoPrism send less data? 40Mb/s sent surely won't be an issue when you have 200Mb/s internet available on both sides...
I recorder a video with my server performance and sunced it (more or less) with what I see on my smartphone. The server sends 40Mb/s when watching and video chunks on the phone

https://streamable.com/ttgqxk

**Comment by lastzero** [comment]
Do you use a VPN / CDN like Cloudflare?

**Reply by lastzero** [reply, marked as answer]
Note that MPEG-4 AVC videos are not re-encoded if they exceed the configured bitrate limit. To reduce the size of AVC videos, you can manually replace the original files with a smaller version or wait for a future release that offers this functionality.

**Reply by lastzero** [reply]
👉 https://docs.photoprism.app/getting-started/advanced/transcoding/#bitrate-limiting

**Reply by shadow82** [reply]
> Do you use a VPN / CDN like Cloudflare?

No - From performance point of view, the only thing a VPN can do for you is slow you down

**Reply by lastzero** [reply]
https://docs.photoprism.app/getting-started/using-a-cdn/

**Reply by shadow82** [reply]
> https://docs.photoprism.app/getting-started/using-a-cdn/

Well.. CDN AFAiK is a solution that offers you content closeby, even normal original content would be far far away.
For instance - assuming that FB keeps it's servers in US only, EU users might use it fast, thanks to CDNs located in Amsterdam and Frankfurt.

But CDN won't solve anything if your server and client is in the same city and same ISP

But thanks for hints :)

As for now I converted problemating video with Freemake Video Converter. It decreased quite non loosely 100MB video into 10MB and the server now sends around 10Mb/s when watching. Video is smooth.

Now I need to find good CLI based converter tool and make some script watching through galleries folder for non converted videos.
This gets closer and closer to private cloud than self hosted photos :D

**Reply by lastzero** [reply]
If you use a proxy or router, you should check if there's a rate or speed limit configured. Networking is complex. A CDN would help if the uplink is too slow and you are not in the same network, even if you are not extremely far away.

**Reply by shadow82** [reply]
> If you use a proxy or router, you should check if there's a rate or speed limit configured. Networking is complex. A CDN would help if the uplink is too slow and you are not in the same network, even if you are not extremely far away.

Everytone use a router - it's a home gateway you have from your ISP.
I know something about networks, since I work with them since 2009 :D 

And a proxy is something I need to add on my homeserver in order not to terminate any connections directly on Linux where PhotoPrism is running. Any 0day or other flaw and "russian hacker" has access to it.
Better to HA Proxy than PhotoPrism and other stuff

**Reply by lastzero** [reply]
Just saying it's a lot of guesswork for us. We previously has users eg with a nginx or Traefik configuration that interfered with the API performance. If that is not the case for you, this is great and you might just have to reduce video bitrate. You can use ffmpeg in a terminal to do this.

**Reply by lastzero** [reply]
> Better to HA Proxy than PhotoPrism and other stuff

HAProxy? Check if others have similar issues, for example:

- https://github.com/OctoPrint/OctoPrint/issues/1846

**Reply by shadow82** [reply]
> > Better to HA Proxy than PhotoPrism and other stuff
> 
> HAProxy? Check if others have similar issues, for example:
> 
> * [HaProxy tremendously slowing down Webcam Stream Performance OctoPrint/OctoPrint#1846](https://github.com/OctoPrint/OctoPrint/issues/1846)

I said I want to deploy HA Proxy at my homelab, but I don't use it yet.
More problems are still to come ;-)
